content/sklearn/concepts/clustering/clustering.md

```diff
@@ -0,0 +1,105 @@
+  - 'Unsupervised learning'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/data-science'
+---
+
+**Clustering** is an unsupervised machine learning technique. It groups similar rows of unlabled data. There are various clustering algorithms (like k-means,DBSCAN etc.). Different clustering algorithms apply different type of clustering. We will only be discussing the algorithms and types of clustering that can be implemented using sklearn.
+
+Clustering is more used as an exploratory data analysis.Each technique varies in their understanding of what a cluster constitutes and how to find them. So, for a dataset each algorithm may come up with different clusters. Since there is no truth against to verify, it's important to check the result. Until the quality of output is satisfactory, iterating to find the clusters are important.
+
+Each clusters are formed based on the similarity to each other. When the dataset has more and complex features the similarity measure becomes complex. After clustering, each cluster is given a unique cluster ID.
+
+Each clustering algorithm comes in two forms- a class and a function. The class implements the fit method to learn the clusters on training data. The function returns an array of integer labels corresponding to different clusters. 
+
+The 12 algorithms that can be executed using sklearn clustering are k-means,Affinity Propagation, Mean shift, Spectral clustering, Ward hierarchical clustering, Agglomerative clustering, DBSCAN, HDBSCAN, OPTICS, Gaussian mixtures, BIRCH, Bisecting k-means.
+
+## K-means
+
+K-means algorithm is a popular clustering algorithm. Given a dataset with n points, the goal is to choose k centers so as to minimize the inertia. 
+
+First, we choose the number of centroids. The algorithm consists of two steps. The first step is to assign each sample to its nearest centroid. Measure the distance between the data point and the current data center via Euclidean distance for instance. The second step is to calculate the mean of all the data points assigned to each cluster. This will be the new cluster center.These two steps are continuously repeated. The algorithm stops when the difference between the old centroid and new centroid is less than the threshold.
+
+## Clustering Performance Evaluation
+
+Evaluating the clusters is not simple like counting errors or precision or recall, simply because there is no ground truth. We can check how similar the data points of a cluster are to each other based on some ground truth instead of looking at the labels.
+
+Visualling look at the clusters, Silhouette Score, rand index and homogeneity are among the few methods to evaluate the clusters.
+
+## Syntax
+Sklearn provides the `KMeans` class for implementing clustering.
+
+```psuedo
+KMeans(n_clusters: int = 8, *, init: str = "k-means++", n_init: str = "auto", max_iter: int = 300, tol: float = 0.0001, verbose: int = 0, random_state: Any | None = None, copy_x: bool = True, algorithm: str = "lloyd") 
```

Comment:
This can be confusing for some readers, can we re-write it in the following format:

```
KMeans(n_clusters=8, init=’k-means++’, n_init=10, max_iter=300, tol=0.0001, precompute_distances=’auto’, verbose=0, random_state=None, copy_x=True, n_jobs=1, algorithm=’auto’)
```